src/parser/xmlHandler.js

```diff
@@ -60,6 +60,7 @@ class XMLHandler {
       name = descriptor.qname.name;
       let isSimple = descriptor.isSimple;
       let attrs = null;
+      if (val === null && descriptor.isNillable) descriptor.type.name = 'string';
```

Comment:
This is hacky. Can we fix https://github.com/strongloop/strong-soap/blob/f7430739366af6a24f1fdede0d1c6ec27c6a073f/src/parser/xmlHandler.js#L881 to return `null` if `val == null`?

Reply:
I originally looked  at that, but there are other date things that are going on before the code gets to the line to check for null and nilable.  I do agree that this seems sort of hacky, but the more I thought about it the more I came to the conclusion that if the field is nilable and the value is null, the code should really not care about the data type from there on out.

Reply:
I agree. Can we refactor the code a bit to cover `nillable` date?

Reply:
My apologies, it looks like getting toSmlDateOrTime to pass back null did work, not sure what I did different the first time I tried it.

Reply:
Great. Can you revert this change then?

Reply:
Done (I thought I has already but...well, it seems I am having a hell of a week).